deps/rabbit/src/rabbit_stream_queue.erl

```diff
@@ -491,6 +494,50 @@ i(type, _) ->
     stream;
 i(_, _) ->
     ''.
+-spec status(rabbit_types:vhost(), Name :: rabbit_misc:resource_name()) ->
+    [[{binary(), term()}]] | {error, term()}.
+status(Vhost, QueueName) ->
+    %% Handle not found queues
+    QName = #resource{virtual_host = Vhost, name = QueueName, kind = queue},
+    case rabbit_amqqueue:lookup(QName) of
+        {ok, Q} when ?amqqueue_is_classic(Q) ->
+            {error, classic_queue_not_supported};
+        {ok, Q} when ?amqqueue_is_quorum(Q) ->
+            {error, quorum_queue_not_supported};
+        {ok, Q} when ?amqqueue_is_stream(Q) ->
+            Data = osiris_counters:overview(),
+            case maps:get({osiris_writer, QName}, Data, undefined) of
+                undefined ->
+                    #{};
+                #{segments := Segments} = Map ->
+                    Conf = amqqueue:get_type_state(Q),
+                    Max = maps:get(max_segment_size, Conf, osiris_log:get_default_max_segment_size()),
+                    Map#{max_disk_size => Segments * Max}
```

Comment:
Maybe we should call this something like `approx_disk_size` or something like that.